crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -233,7 +233,9 @@ impl PgManager {
             .or(before)
             .map(|cursor| self.parse_obj_cursor(&cursor))
             .transpose()?;
-
+        let coin_type = parse_to_type_tag(Some(coin_type))
+            .map_err(|e| Error::Internal(e.to_string()))?
```

Comment:
can you use Error::InvalidCoinType

Reply:
Note that in some cases, this is an internal error, and in some cases this is a user error -- it depends on the callee, this is why the DB layer needs its own error type.

Reply:
Until it does, I'm OK with us using either error type here -- neither is ideal, unfortunately.

Reply:
ok that's fair
we have a DbValidationError enum that we could add InvalidCoinType to

Reply:
<img width="2539" alt="image" src="https://github.com/MystenLabs/sui/assets/135084671/bd568987-0ce5-4f8a-8675-c4077025f920">